otherlibs/unix/caml/socketaddr.h

```diff
@@ -92,6 +110,25 @@ extern value caml_unix_alloc_inet6_addr (struct in6_addr * inaddr);
 
 #ifdef __cplusplus
 }
+
+/* API compatibility between union sock_addr_union and
+   struct sockaddr_storage. C++ doesn't support C11 _Generic. */
+static inline void
+caml_unix_get_sockaddr(value vaddr, union sock_addr_union * addr /*out*/,
+                       socklen_param_type * addr_len /*out*/) {
+  caml_unix_get_sockaddr(vaddr,
+                         reinterpret_cast<struct sockaddr_storage *>(addr),
+                         addr_len);
+}
+static inline value
+caml_unix_alloc_sockaddr(union sock_addr_union * addr /*out*/,
+                         socklen_param_type addr_len /*out*/,
+                         int close_on_error) {
+  return caml_unix_alloc_sockaddr(
+    reinterpret_cast<struct sockaddr *>(addr),
+    addr_len,
+    close_on_error);
+}
```

Comment:
I think it would make it more legible to move that in an `#else` just after the C version, instead of hiding it with the `external "C"` block end.

How about saying in the comment: `C++ doesn't support C11 _Generic, so use function overloading.`?